lib/Rex/DetectCLI.pm

```diff
@@ -0,0 +1,89 @@
+#
+# (c) Jan Gehring <jan.gehring@gmail.com>
+#
+# vim: set ts=2 sw=2 tw=0:
+# vim: set expandtab:
+
+package Rex::DetectCLI;
+
+require Exporter;
+use base qw(Exporter);
+use vars qw(@EXPORT);
+
+use List::MoreUtils qw'firstidx';
+use Data::Dumper;
+
+use PPI;
+
+@EXPORT = qw(detect_cli);
+
+sub detect_cli {
```

Comment:
This is pretty cool.  I mean its also insane.  I don't love it.  But I don't have any better ideas atm.  I thought PPI would be really slow.  Its not as bad as I thought though...